pkg/cache/v3/linear.go

```diff
@@ -21,46 +21,73 @@ import (
 	"strconv"
 	"strings"
 	"sync"
-	"sync/atomic"
 
 	"github.com/envoyproxy/go-control-plane/pkg/cache/types"
 	"github.com/envoyproxy/go-control-plane/pkg/log"
 )
 
+// cachedResource is used to track resources added by the user in the cache.
+// It contains the resource itself and its associated version (currently in two different modes).
+// TODO(valerian-roche): store serialized resource if already marshaled to compute the version.
```

Comment:
This rework is made quite complicated by the current proliferation of usage of the internal RawResponse within the different packages test code.